Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp

```diff
@@ -83,6 +80,29 @@ struct DMABufFeedback {
         {
         }
 
+        explicit FormatTable() = default;
+        explicit FormatTable(uint32_t size, int fd)
+            : size(size)
+            , data(static_cast<FormatTable::Data*>(mmap(nullptr, size, PROT_READ, MAP_PRIVATE, fd, 0)))
+        {
+            if (data == MAP_FAILED) {
+                data = nullptr;
+                size = 0;
+            }
+        }
+
+        explicit operator bool() const
+        {
+            return size && data;
+        }
+
+        FormatTable& operator=(FormatTable&& other)
+        {
+            std::swap(size, other.size);
+            std::swap(data, other.data);
```

Comment:
Why swap? what we want here is that the moved object is reset after the move. That will always happen for the move constructor, but for the assignment operator, the object might have a value that we don't want in the moved object. I think std::exchange is what we want here, or am I missing something?

Reply:
If `this->data` was valid we need to `munmap()` it. The swap is there to make the possibly valid values of `data` and `size` end up in `other`: the destructor will be called on `other` after the function returns because the compiler knows `other` cannot be used further because it was moved into the function call. The destructor on `other` will call `munmap()` for the previous value of `this->data`. I understand this is a common RAII idiom.

Reply:
Then, let's do the unmap here (but only if the map is different, of course). We need to make sure that after committed = WTFMove(pending); pending is reset, we don't want to swap them unless committed is already in a reset state.